server/graphql/common/expenses.ts

```diff
@@ -244,6 +245,14 @@ export const canPayExpense = async (req: express.Request, expense: typeof models
   } else if (!canUseFeature(req.remoteUser, FEATURE.USE_EXPENSES)) {
     return false;
   } else {
+    const collective = expense.collective || (await models.Collective.findByPk(expense.CollectiveId));
```

Comment:
We should try to only use loaders in these helpers, as they're massively used in lists. Same for loading the host and expense approver below

```suggestion
    const collective = expense.collective || (await req.loaders.Collective.byId.load(expense.CollectiveId));
```

Maybe we can create a new loader (e.g. `req.loaders.Expense.getHost`) by extracting the code from https://github.com/opencollective/opencollective-api/blob/917c228c5ae0b086cb2d6a9a2c09a93d51e3c247/server/graphql/v2/object/Expense.js#L154-L161?

Reply:
I agree, I'll abstract the host loader.